src/pages/PageLayout.tsx

```diff
@@ -1,39 +1,148 @@
-  pages,
-  title,
-  children,
-  ...restProps
-}) => {
-  useDocumentTitle(`Threshold - ${title}`)
-  const links = pages
-    ? pages.filter((_) => !_.route.hideFromMenu).map((_) => _.route)
-    : []
+const getBackgroundStyles = (variant: VariantType): SystemStyleObject => ({
+  bgImage:
+    variant === "primary" ? PRIMARY_BACKGROUND_PATH : SECONDARY_BACKGROUND_PATH,
+  bgRepeat: "no-repeat",
+  bgSize: variant === "primary" ? "contain" : "cover",
+  bgPosition: "bottom",
+  bgAttachment: "fixed",
+})
+
+const FixedContainer: FC<FlexProps> = (props) => (
+  <Flex flexFlow={"column"} maxW={"1920px"} mx={"auto"} {...props} />
+)
+
+/**
+ * Renders the page wrapped in layout with render slots for content alignment.
+ * @param {PageLayoutProps} props - The props for the PageLayout component.
+ * @return {JSX.Element} The rendered page wrapped in layout.
+ */
+const PageLayout: FC<PageLayoutProps> = (props) => {
+  const {
+    renderTop,
+    renderLeft,
+    renderRight,
+    backgroundVariant = "primary",
+    children,
+    ...restProps
+  } = props
+
+  const [hasContentMaxWidth] = useMediaQuery(
+    `(min-width: ${CONTENT_MAX_WIDTH})`
+  )
+
+  const childrenContainerColumnSpan = [
+    renderLeft === null || renderLeft !== undefined ? 2 : 1,
+    renderRight === null || renderRight !== undefined ? 4 : 5,
+  ].join("/")
+
+  const borderX = {
+    borderLeft: renderLeft !== undefined ? BORDER : undefined,
+    borderRight: renderRight !== undefined ? BORDER : undefined,
+  }
 
   return (
-    <>
-      {links.length > 0 && <SubNavigationPills links={links} />}
-      <Container
-        maxW={{ base: "2xl", xl: "6xl" }}
-        mt="6.25rem"
-        my={16}
-        {...restProps}
-      >
-        {children}
-        <Outlet />
-      </Container>
-    </>
+    <Flex
+      flexFlow={"column"}
+      bg={"black"}
+      color={"white"}
+      sx={getBackgroundStyles(backgroundVariant)}
+      minH={"100vh"}
+      {...restProps}
+    >
+      <Header isDesktopViewport={hasContentMaxWidth} />
+      {!!renderTop && (
+        <Box p={6} borderBottom={BORDER}>
+          <FixedContainer
+            mx="auto"
+            w="full"
+            maxW={`calc(${CONTENT_MAX_WIDTH} + 2 * ${
+              hasContentMaxWidth ? "0px" : spacing[6]
+            })`}
+          >
+            {renderTop}
+          </FixedContainer>
+        </Box>
+      )}
+      <FixedContainer flex={1} w={"full"}>
+        <Grid
+          templateColumns={{ base: "auto", lg: "repeat(4, .5fr)" }}
+          w="full"
+          maxW={CONTENT_MAX_WIDTH}
+          mx="auto"
+          {...borderX}
+          flex={1}
+        >
+          {renderLeft && (
+            <Box p={hasContentMaxWidth ? 10 : 6}>{renderLeft}</Box>
+          )}
+          <Box
+            order={{ base: -1, lg: "unset" }}
+            gridColumn={{ base: "auto", lg: childrenContainerColumnSpan }}
+            {...borderX}
+          >
+            {children}
+          </Box>
```

Comment:
I think we should add some padding to this box by default.

<img width="326" alt="image" src="https://github.com/threshold-network/bitcoin-on-base/assets/40306834/985d9209-b987-413c-bf20-eb9997638d08">

Reply:
There is a scenario with minting success page where the background element will be missplaced if the padding will be applied here.
I will introduce additional prop to conditonally render this padding around children.
<img width="783" alt="image" src="https://github.com/threshold-network/bitcoin-on-base/assets/11503879/5d5a995b-35d9-4e26-aef4-faa81ceefdd1">